### DISCUSSION TITLE
Issue with Scripts Running Slowly and Getting Stuck After Updating SeleniumBase and Pip

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2888] [Q&A] [answered]
Hi,

I'm encountering issues with my SeleniumBase scripts after updating to the latest version (4.28.2) and upgrading pip. The scripts are running very slowly, often failing to initialize the driver, and sometimes getting stuck without even opening the browser. Here's what I've done so far:

Created a new virtual environment.
Installed all necessary libraries.
Attempted to run a basic script to ensure functionality:

`from seleniumbase import SB

with SB(uc=True) as sb:
    sb.open("https://www.google.com")
    sb.type('input[name="q"]', "SeleniumBase")
    sb.click('input[type="submit"]')
    sb.assert_text("SeleniumBase", "h3")
`

Even with this basic script, the execution either hangs or runs extremely slowly.

Additionally, I've encountered issues with deleting some folders in my environment, specifically _tests, which seems to be causing some problems.

Could this be related to the recent updates? Is there a known issue, or do you have any suggestions for resolving these performance problems?

Thank you!

**Comment by mdmintz** [comment, marked as answer]
It's a `textarea` (not an `input`) for the Google Search Box. (Or they alternate with an A/B test).
Therefore, your script would sit for 10 seconds, (the default `timeout` for `sb.type()`), and then fail.

Here's the correct script:

```python
from seleniumbase import SB

with SB(uc=True) as sb:
    sb.open("https://www.google.com")
    sb.type('[name="q"]', "SeleniumBase")
    sb.click('input[type="submit"]')
    sb.assert_text("SeleniumBase", "h3")
```

**Reply by benfrankie** [reply]
I am facing an issue where my scripts are running slowly and getting stuck after updating SeleniumBase and pip. Initially, a simple test script runs fine, but after launching a few more scripts, even the test script starts to hang. Here’s a summary of what I’ve tried and the issues I am facing:

Issue Summary:

Scripts run fine initially but get stuck after a while.
Test script hangs after running a few other scripts.
Updated SeleniumBase and pip recently.
Ensured correct locators as per your suggestion (using [name="q"] for the Google search box).
Environment:

Running on Windows.
Using seleniumbase version 4.28.2.
Encountered RuntimeError: Numpy is not available which was resolved by downgrading numpy.
Steps Taken:

Updated and reinstalled dependencies.
Corrected element locators as suggested.
Attempted to isolate and run scripts one by one.
Despite these efforts, the issue persists. Could you please provide further guidance on how to ensure stability and performance of the scripts? Are there any specific configurations or settings that might help resolve this?

Thank you for your help!

**Reply by mdmintz** [reply]
Upgrade to `4.28.3` (Fixes https://github.com/seleniumbase/SeleniumBase/issues/2889)

**Reply by benfrankie** [reply]
If possible, how can I use the sb.uc_gui_handle_cf() function on Windows in headless mode?
I encounter the error: 'Error while handling Turnstile challenge: PyAutoGUI can't be used in headless mode'.

Thank you!

**Reply by mdmintz** [reply]
`pyautogui` cannot click on invisible elements. Therefore, headless mode is not possible if `pyautogui` is being used.
